add ovn to shiftstack-ci presubmits

jluhrsen opened the pull request:
fix test flake that checks if the job name matches the running CNI, which is ovnk in 4.12+

jobs were found using this search:

❯ w3m -dump -cols 200
"https://search.ci.openshift.org/?maxAge=96h&type=junit&search=job+name.*does+not+have+network+type+in+name.*expected.*ovn" | sed -n 's/^\([^ ]*\) .*failures match.*/\1/p' | rg 'shiftstack' periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.12-e2e-openstack-parallel periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.12-e2e-openstack-serial periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.12-e2e-openstack-appcreds

Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

jluhrsen commented:
https://issues.redhat.com/browse/SDN-3460

jluhrsen commented:
/test pj-rehearse

jluhrsen commented:
/test pj-rehearse

jluhrsen commented:
/assign @bparees 

all the rehearsals failed, but the CNI flakes are fixed with this PR.

bparees commented:
/approve
/lgtm

pierreprinetti commented:
@jluhrsen 

Despite what its description states, this PR is doing nothing like “fixing a test flake”. I have no idea what test it’s making flake, and the Jira card you link is no more explicit.

Instead, this PR is enforcing some rule about test naming that I am hearing about here for the first time. And removing our Slack reporting in the process.

A communication to the team on Slack, and a pointer to the conventions you want to enforce  (and their rationale), would have been appreciated. We are happy to collaborate with the teams that support our work, but such changes don’t help.

CC @bparees @mandre

bparees commented:
> Despite what its description states, this PR is doing nothing like “fixing a test flake”. I have no idea what test it’s making flake, and the Jira card you link is no more explicit.

The PR description provides more detail and links to an example of what is failing(flaking) and why these job renames are necessary/what problem they are fixing:

```
# [sig-ci] [Early] prow job name should match network type [apigroup:config.openshift.io] [Suite:openshift/conformance/parallel]
flake: job name "upgrade-aws" does not have network type in name (expected `ovn`)
Ginkgo exit error 4: exit with code 4
```

(not the failure for this specific job name, but it applies to any job that runs this test)

> Instead, this PR is enforcing some rule about test naming that I am hearing about here for the first time. And removing our Slack reporting in the process.
> A communication to the team on Slack, and a pointer to the conventions you want to enforce (and their rationale), would have been appreciated. We are happy to collaborate with the teams that support our work, but such changes don’t help.

Apologies for how this impacted your team. @jluhrsen has been working through all our CI jobs to get the naming conventions straightened out, there are hundreds of jobs to cover and in most cases there's no need to involve teams in the change (even assuming it's easy to figure out which team, if any, owns the job in question).

With regards to removing the slack notification, absolutely this shouldn't have happened and I should have reviewed the changes more carefully.  That said, again, we've been doing a bunch of these PRs and the diffs are usually quite messy because the job renames result in reordering of generated content making it not obvious when customizations have been stripped out.  So I hope you can appreciate that accidents happen and this will get fixed.

I apologize for the impact this had on your team, and i'm glad you caught it.  We'll get your slack notifications back in place asap.

bparees commented:
@pierreprinetti restoring the slack notifications here, ptal: https://github.com/openshift/release/pull/34073

pretty sure i got them all, but please double check me.  All the 4.12 jobs in that PR have notifications configured.

@jluhrsen ptal as well

pierreprinetti commented:
Thanks @bparees ; I was going to do that.

You have no reason to apologise; I am just worried about changes being made to our test batteries.

jluhrsen commented:
> Thanks @bparees ; I was going to do that.
> 
> You have no reason to apologise; I am just worried about changes being made to our test batteries.

well, I apologize as well anyway :)  I've created upwards of 50 PRs for this same work in all sorts
of places and sometimes it's hard to figure out which slack channel (there is no .*shiftstack.* channel
I could find) or which OWNERS file people I can ping (names in OWNERS don't always match up to
slack usernames), but I did check. @bparees is my fall-back in those cases.

I'll have to keep a closer lookout for this. I always make a bad assumption that `make jobs` is not going
to remove something that I haven't explicitly changed in the config/ yaml. which is wrong :(


> @pierreprinetti restoring the slack notifications here, ptal: #34073
> 
> pretty sure i got them all, but please double check me. All the 4.12 jobs in that PR have notifications configured.
> 
> @jluhrsen ptal as well

Thank you. I see your PR is merged. That took care of 4.12. Here's [another one](https://github.com/openshift/release/pull/34081) to take
care of 4.13.

It seems almost all jobs in this shiftstack-ci have the shiftstack-bot reporter, but not all. Here's
a summary of what I found while I was double checking. @pierreprinetti, in case you were
curious.

```
   1   │   name: periodic-ci-shiftstack-shiftstack-ci-main-monitor-mecha-az0
   2   │   name: periodic-ci-shiftstack-shiftstack-ci-main-monitor-mecha-central
   3   │   name: periodic-ci-shiftstack-shiftstack-ci-main-monitor-mecha-kuryr
   4   │   name: periodic-ci-shiftstack-shiftstack-ci-main-monitor-nerc-dev
   5   │   name: periodic-ci-shiftstack-shiftstack-ci-main-monitor-openstack-hwoffload
   6   │   name: periodic-ci-shiftstack-shiftstack-ci-main-monitor-openstack-nfv
   7   │   name: periodic-ci-shiftstack-shiftstack-ci-main-monitor-vexxhost
   8   │   name: periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.10-upgrade-from-stable-4.9-e2e-openstack-upgrade
   9   │   name: periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-ovn-appcreds
  10   │   name: periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-ovn-parallel
  11   │   name: periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-ovn-serial
  12   │   name: periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.7-upgrade-from-stable-4.6-e2e-openstack-upgrade
  13   │   name: periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.8-upgrade-from-stable-4.7-e2e-openstack-upgrade
  14   │   name: periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.9-upgrade-from-stable-4.8-e2e-openstack-upgrade
```
 
Note: the 4.13 jobs above will be fixed with [my PR](https://github.com/openshift/release/pull/34081).


2) the recently added 4.12->4.13 openstack upgrade job did not have the -sdn- string in the job name. I
added that in the same PR that puts reporters back in the 4.13 jobs. I made sure the reporter didn't get
removed this time.

pierreprinetti commented:
Thank you @jluhrsen .
